From af15632601d7cab98877c37787a7e84e3b9e3506 Mon Sep 17 00:00:00 2001
From: Sughosh Ganu <sughosh.ganu@linaro.org>
Date: Thu, 8 Apr 2021 12:30:55 +0530
Subject: [PATCH 1/1] efi_loader: efi_esrt: Fix the build warning for 32 bit
 systems
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fix the build warning when building for 32 bit systems by using the
length modifier for size_t.

lib/efi_loader/efi_esrt.c: In function ‘efi_esrt_populate’:
include/efi_loader.h:126:8: warning: format ‘%ld’ expects argument of type ‘long int’, but argument 8 has type ‘size_t’ {aka ‘unsigned int’} [-Wformat=]

Signed-off-by: Sughosh Ganu <sughosh.ganu@linaro.org>
Reviewed-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 lib/efi_loader/efi_esrt.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/efi_loader/efi_esrt.c b/lib/efi_loader/efi_esrt.c
index 947bdb5e95..8815e56e15 100644
--- a/lib/efi_loader/efi_esrt.c
+++ b/lib/efi_loader/efi_esrt.c
@@ -341,7 +341,7 @@ efi_status_t efi_esrt_populate(void)
 		return EFI_SUCCESS;
 	}
 
-	EFI_PRINT("ESRT populate esrt from (%ld) available FMP handles\n",
+	EFI_PRINT("ESRT populate esrt from (%zd) available FMP handles\n",
 		  no_handles);
 
 	/*
-- 
2.30.2

